tests/trilinos/nox_solver_01.cc

```diff
@@ -0,0 +1,222 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// Check TrilinosWrappers::NOXSolver by solving f(x) = x^2 with initial
+// condition x=2.
```

Comment:
Do you mean to say that you're solving `f(x)=0` with `f(x)=x^2`? That's not a great test case (or maybe it actually is!) because the Jacobian is singular at the solution.

Reply:
This is what I solve.